### PR TITLE
Remove `WildcardPattern.typeAnnotation`

### PR DESCRIPTION
**Pull request by rintaro**
Type annotation has never been a part of `wildcard-pattern`.

### CODE REVIEW

**Comment by rintaro** [comment]
@swift-ci Please test

File: Sources/SwiftSyntax/SwiftSyntaxCompatibility.swift [hunk 1]
```diff
@@ -509,6 +509,30 @@ public extension TypeEffectSpecifiersSyntax {
...
+      nil,
+      trailingTrivia: trailingTrivia
+    );
+  }
+
+  @available(*, deprecated, message: "'typeAnnotation' was removed")
+  var typeAnnotation: TypeAnnotationSyntax? {
+    get { nil }
+    set {}
+  }
```
Comment (ahoppen) [inline Sources/SwiftSyntax/SwiftSyntaxCompatibility.swift:533]: I think we should also cover the unexpected children in the compatibility layer. See https://github.com/apple/swift-syntax/pull/2379#discussion_r1421197690

**Review by ahoppen — COMMENTED** [review]
I think it would be good to add a release note for this change + add a test case so we don’t make the same mistake again 😉

**Comment by rintaro** [comment]
@ahoppen  I'm going to get this into 5.10. What's the manner for wring the release notes? Is it OK to write it directly in 5.10 section before the approval?

**Comment by ahoppen** [comment]
Yes, that’s OK

**Review by ahoppen — APPROVED** [review]

**Comment by rintaro** [comment]
@swift-ci Please test macOS

**Comment by rintaro** [comment]
@swift-ci Please test Linux

**Pull request merged**